pkg/services/publicdashboards/queries/dashboard_queries_test.go

```diff
@@ -458,6 +532,27 @@ func TestGroupQueriesByPanelId(t *testing.T) {
             "refId": "A"
 		}`, string(query))
 	})
+
+	t.Run("hidden query filtered", func(t *testing.T) {
+		json, err := simplejson.NewJson([]byte(dashboardWithOneHiddenQuery))
+		require.NoError(t, err)
+		queries := GroupQueriesByPanelId(json)[2]
+
+		require.Len(t, queries, 1)
+		for _, query := range queries {
+			if hideAttr, exists := query.CheckGet("hide"); exists && hideAttr.MustBool() {
```

Comment:
Is it possible to loop and match on a specific key of the hidden query to make sure it's not in the collection instead of this check?

Reply:
what do you mean? didn't get you